src/app/pages/history/history.component.ts

```diff
@@ -78,6 +83,8 @@ export class HistoryComponent implements OnInit, OnDestroy {
   }
 
   private getDataChart(): void {
+    this.preDataChart = [];
+    this.dataChart = [];
     this.dataTable.forEach((item: IEvents) => this.preDataChart.push({name: item.category, y: +item.amount}));
     if (this.dataChart.length !== 0) {
```

Comment:
so you did that _this.dataChart = [];_ and after that you check that _this.dataChart.length !== 0_ 
how it must work?

Reply:
Like with the tables, I clear all massive before using the method again to prevent item doubling